WordPress/WordPressTest/FollowCommentsServiceTests.swift

```diff
@@ -0,0 +1,97 @@
+import XCTest
+import WordPressKit
+
+@testable import WordPress
+
+class FollowCommentsServiceTests: XCTestCase {
+
+    private var contextManager: TestContextManager!
+
+    // MARK: - Lifecycle
+
+    override func setUp() {
+        super.setUp()
+        contextManager = TestContextManager()
+    }
+
+    override func tearDown() {
+        ContextManager.overrideSharedInstance(nil)
+        super.tearDown()
+    }
+
+    // MARK: - Mocks / Tester
+
+    class ReaderPostServiceRemoteMock: ReaderPostServiceRemote {
+
+        var subscribeToPostCalled = false
+        var unsubscribeFromPostCalled = false
+
+        override func subscribeToPost(with postID: Int,
+                                      for siteID: Int,
+                                      success: @escaping () -> Void,
+                                      failure: @escaping (Error?) -> Void) {
+            subscribeToPostCalled = true
+        }
+
+        override func unsubscribeFromPost(with postID: Int,
+                                          for siteID: Int,
+                                          success: @escaping () -> Void,
+                                          failure: @escaping (Error) -> Void) {
+            unsubscribeFromPostCalled = true
+        }
+    }
+
+    class FollowCommentsServiceTester: FollowCommentsService {
```

Comment:
Since `FollowCommentsService` is the object which we're testing here I'd try to avoid inherit from it.

What you can do instead is to inject the service into the class. There are two distinct, yet similar, ways you can accomplish that:

1. https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/Services/PostCoordinator.swift#L46
2. https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/Services/ReaderCardService.swift#L20

Both ways will allow you to inject your `ReaderPostServiceRemoteMock` without the need of creating a "Tester". :)

Reply:
@leandroalonso 
Thanks for the suggestions! I'll give this a shot 👍

Reply:
It seems like Objective-C doesn't support Swift default parameters. So no matter what, it seems like you'll have to pass an argument for the service parameter... which is unfortunate, because that means we won't be able to leverage the convenience of default paramters.

While the default parameter approach works well for the tests, I think it looks a little unwieldy when called from Objective-C. What do you think?

### Approach 1
```
FollowCommentsService *service = [[FollowCommentsService alloc] initWithPost:(ReaderPost * _Nonnull) service:(ReaderPostServiceRemote * _Nullable)];

```

![Screen Shot 2020-08-20 at 21 28 35](https://user-images.githubusercontent.com/6711616/90772263-6541ca80-e32f-11ea-8bbc-912e7dd73dc5.png)


### Approach 2
```
FollowCommentsService *service = [[FollowCommentsService alloc] initWithPost:(ReaderPost * _Nonnull) service:(ReaderPostServiceRemote * _Nonnull)];
```

![Screen Shot 2020-08-20 at 21 42 30](https://user-images.githubusercontent.com/6711616/90771638-8fdf5380-e32e-11ea-95e9-40ead8baeff4.png)

Reply:
Hmmm, I see... 🤔

What if we create a `class func` that acts as the constructor and use that in the Obj-C side? That way we can still inject it in the tests and use the "custom constructor" in Obj-C. Thoughts?

Reply:
That's a great idea! 👍

Reply:
Fixed! 5b90251
The dependency injection approach is much cleaner 😄